### DISCUSSION TITLE
Faust.js & NextJS Preview Mode

### DISCUSSION THREAD

**Discussion by carlassmann** [wpengine/faustjs#709] [Ideas]
I am looking into FaustJS as I want to enable preview mode on my WordPress site. NextJS [has a preview Mode](https://nextjs.org/docs/advanced-features/preview-mode) which follows the following procedure:

1. The CMS allows for a custom preview URL which will look like this: `https://<your-site>/api/preview?secret=<token>&slug=<path>`
2. My NextJS APi route checks the validity of the `<token>`, sets a preview context and redirects to `<path>`
3. The route generates the preview just like it would generate the default page, but with preview data.
4. The user sees the preview page.

FaustJS seems to render ALL previews at a specific route while following an OAuth flow. I prefer sticking to the NextJS way of handling this.

My question: Is there a chance to configure the FaustWP Plugin to allow me following the way, NextJS handles previews? If not, is support for this approach planned in the future?

**Comment by blakewilson** [comment]
@ccssmnn Right now there is no support for Next.js Preview Mode. However, we are looking into ways to support it. I'll write back here when we have more info!

**Comment by dawnerd** [comment]
I'd also love to see this. The current setup doesn't even really work for me. Been evaluating headless systems for use with some client sites and I know they're going to want to be able to click around while in a preview mode.
